api/users/views.py

```diff
@@ -63,7 +64,18 @@ def get_user(self, check_permissions=True):
         return obj
 
 
-class UserList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
+class UserFilter(NewDjangoFilterMixin):
+
+    full_name = django_filters.CharFilter(name='fullname', lookup_expr='icontains')
+    id = django_filters.CharFilter(name='guids___id')
+
+    class Meta:
+        model = OSFUser
+        fields = ['id', 'full_name', 'given_name', 'middle_names', 'family_name']
+        strict = django_filters.constants.STRICTNESS.RAISE_VALIDATION_ERROR
```

Comment:
Also, is there a way for us to raise the same error that we're doing now if the client passes an invalid field name? https://github.com/CenterForOpenScience/osf.io/blob/7e41c06588a5d1e370f2ac2a4b1b6e57e2a6c159/api/base/filters.py#L116